### PR TITLE
Fix loader bug of profile page

### PR DESCRIPTION
**Pull request by Palakkgoyal**
This PR fixes issue no. #883 
I have added the condition for finding user and if not found then show 404

### CODE REVIEW

**Review by narayan954 — APPROVED** [review]
looks good to me!

File: src/pages/Profile/index.jsx [hunk 1]
```diff
@@ -72,12 +73,17 @@ function Profile() {
       docRef
         .get()
         .then((snapshot) => {
-          const doc = snapshot.docs[0];
-          setUserData({
-            name: doc.data().name,
-            avatar: doc.data().photoURL,
-            uid: doc.data().uid,
-          });
+          if(snapshot.exists){
```
Comment (narayan954) [inline src/pages/Profile/index.jsx:76]: This is perhaps not a valid condition and hence profile was being forwarded to 404 page everytime instead.

Comment (narayan954) [reply src/pages/Profile/index.jsx:76]: I had to revert this, sort of, to fix this

**Comment by Palakkgoyal** [comment]
Is it? let me check.

**Comment by narayan954** [comment]
> Is it? let me check.

yes, you may re add this on your localhost and it shouldn't work! ;-;

**Pull request merged**